fendermint/app/src/settings.rs

```diff
@@ -145,6 +152,29 @@ pub fn expand_tilde<P: AsRef<Path>>(path: P) -> PathBuf {
         .unwrap_or(p)
 }
 
+/// A serde deserialization method to deserialize a subnet path string into a [`SubnetID`].
+pub(crate) fn deserialize_subnet_id<'de, D>(deserializer: D) -> anyhow::Result<SubnetID, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    struct SubnetIDVisitor;
+    impl<'de> serde::de::Visitor<'de> for SubnetIDVisitor {
+        type Value = SubnetID;
+
+        fn expecting(&self, formatter: &mut Formatter) -> std::fmt::Result {
+            formatter.write_str("a string")
+        }
+
+        fn visit_str<E>(self, v: &str) -> std::result::Result<Self::Value, E>
+        where
+            E: serde::de::Error,
+        {
+            SubnetID::from_str(v).map_err(E::custom)
+        }
+    }
+    deserializer.deserialize_str(SubnetIDVisitor)
```

Comment:
```suggestion
    let s = String::deserialize(deserializer)?;
    s.parse::<SubnetID>().map_err(|e| D::Error::custom(e))
```

Requires `use serde::de::Error;` at the top. 

I don't see the visitor pattern very often, I see it's powerful, but maybe for this case a map over a known type is more concise?